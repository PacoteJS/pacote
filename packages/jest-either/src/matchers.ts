export interface AsymmetricMatcher {
  asymmetricMatch(other: any): boolean
}

export function isAsymmetricMatcher(
  matcher: any
): matcher is AsymmetricMatcher {
  return typeof matcher.asymmetricMatch === 'function'
}
