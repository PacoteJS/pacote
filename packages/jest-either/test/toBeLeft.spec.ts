import { left, right } from 'fp-ts/lib/Either'
import matchers from '../src'

expect.extend(matchers)

describe('.toBeLeft()', () => {
  test('passes when object is a left', () => {
    expect(left(true)).toBeLeft()
  })

  test('fails when object is a right', () => {
    expect(() => expect(right(true)).toBeLeft()).toThrowErrorMatchingSnapshot()
  })
})

describe('.not.toBeLeft()', () => {
  test('passes when object is a right', () => {
    expect(right(true)).not.toBeLeft()
  })

  test('fails when object is a left', () => {
    expect(() =>
      expect(left(false)).not.toBeLeft()
    ).toThrowErrorMatchingSnapshot()
  })
})
