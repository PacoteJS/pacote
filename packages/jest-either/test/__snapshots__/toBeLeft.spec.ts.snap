// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toBeLeft() fails when object is a left 1`] = `
"expect(received).not.toBeLeft()

Expected Either not to be left, received left."
`;

exports[`.toBeLeft() fails when object is a right 1`] = `
"expect(received).toBeLeft()

Expected Either to be left, received right."
`;
